### DISCUSSION TITLE
UPnP/Dynamic Port Forwarding via VPN

### DISCUSSION THREAD

**Discussion by Svagtlys** [qdm12/gluetun#2349] [Q&A] [answered]
I've been trying to set up port forwarding through my VPN provider (hide.me), but they do not offer static forwarded ports- every time I launch gluetun, a new port is assigned to my instance. The outbound connection through the port seems to work fine, but inbound connections time out, which my research says is due to firewall issues. I'm hoping to figure out how to allow a port that changes every time through the gluetun firewall, if anyone knows how to help me out? I've tried a few iptables commands, but nothing seems to be working so far.

The iptables commands (added to /iptables/post-rules.txt):
```
iptables-legacy -I INPUT -i tun0 -p tcp -s 10.133.0.0/16 -j ACCEPT
iptables-legacy -I INPUT -i tun0 -p udp -d 10.133.0.0/16 -j ACCEPT
```

where 10.133.0.0/16 encompasses the IPs that the VPN network could provide internally, which show up along with the dynamically assigned port when I've run netstat inside of gluetun to check what port I'm being assigned by my VPN.

**Comment by jagaimoworks** [comment, marked as answer]
You could upvote #1881 to try to get the maintainer to natively support hide.me port forwarding or even #2369, which requests runtime control via the built-in control server.
